subprojects/docs/src/snippets/swift/testReport/groovy/buildSrc/src/main/groovy/myproject.xctest-conventions.gradle

```diff
@@ -0,0 +1,31 @@
+// tag::test-report[]
+plugins {
+    id 'xctest'
+}
+
+xctest {
+    binaries.configureEach {
+        runTask.get().configure {
+            // Disable the test report for the individual test task
+            reports.html.enabled = false
+        }
+    }
+}
+
+// Share the test report data to be aggregated for the whole project
+configurations {
+    binaryTestResultsElements {
+        canBeResolved = false
+        canBeConsumed = true
+        attributes {
+            attribute(Category.CATEGORY_ATTRIBUTE, objects.named(Category, Category.DOCUMENTATION))
+            attribute(DocsType.DOCS_TYPE_ATTRIBUTE, objects.named(DocsType, 'test-report-data'))
+        }
+        afterEvaluate {
+            tasks.withType(XCTest).each {
+                outgoing.artifact(it.binaryResultsDirectory)
+            }
+        }
```

Comment:
`XCTest` tasks are only available `afterEvaluate` 😕

Reply:
If you move this out of `afterEvaluate` the `withType()` is no longer triggered, because the task is only registered in `afterEvaluate`?

Reply:
Looks like it. Without afterEvaluate, `withType(XCTest)` yields an empty collection.

Reply:
Now I see. What if you do `tasks.withType(XCTest) { }` (without the _each_). That should register a configuration closure that is executed lazily the moment the task is registered.

Reply:
This works. Thanks!